operator/v1/types_csi_cluster_driver.go

```diff
@@ -118,25 +121,136 @@ const (
 type CSIDriverConfigSpec struct {
 	// driverType indicates type of CSI driver for which the
 	// driverConfig is being applied to.
-	//
-	// Valid values are:
-	//
-	// * vSphere
-	//
-	// Allows configuration of vsphere CSI driver topology.
-	//
-	// ---
+	// Valid values are: AWS, Azure, GCP, vSphere
 	// Consumers should treat unknown values as a NO-OP.
-	//
 	// +kubebuilder:validation:Required
 	// +unionDiscriminator
 	DriverType CSIDriverType `json:"driverType"`
 
+	// AWS is used to configure the AWS CSI driver.
+	// +optional
+	AWS *AWSCSIDriverConfigSpec `json:"aws,omitempty"`
+
+	// Azure is used to configure the Azure CSI driver.
+	// +optional
+	Azure *AzureCSIDriverConfigSpec `json:"azure,omitempty"`
+
+	// GCP is used to configure the GCP CSI driver.
+	// +optional
+	GCP *GCPCSIDriverConfigSpec `json:"gcp,omitempty"`
+
 	// vsphere is used to configure the vsphere CSI driver.
 	// +optional
 	VSphere *VSphereCSIDriverConfigSpec `json:"vSphere,omitempty"`
 }
 
+// AWSCSIDriverConfigSpec defines properties that can be configured for the AWS CSI driver.
+type AWSCSIDriverConfigSpec struct {
+	// kmsKey sets the cluster default storage class to encrypt volumes with a user-defined KMS key,
+	// rather than the default KMS key used by AWS.
+	// +optional
+	KMSKey AWSKMSKey `json:"kmsKey,omitempty"`
```

Comment:
After flipping around quite a bit in the enhancement, we ultimately landed on having the ARN as a single string rather than structured:
https://github.com/openshift/enhancements/blob/master/enhancements/installer/storage-class-encrypted.md#aws